File: frontend/src/stores/project_category.ts
import { defineStore } from 'pinia'
import {
  getProjectCategoriesApi,
  getProjectCategoryByIDApi,
  createProjectCategoryApi,
  updateProjectCategoryApi,
  deleteProjectCategoryApi,
} from '@/api/project_category'

export interface ProjectCategory {
  id: string
  company_id: string
  name: string
  code?: string
  color?: string
  is_active: boolean
  created_at: string
  updated_at: string
}

export const useProjectCategoryStore = defineStore('projectCategory', {
  state: () => ({
    categories: [] as ProjectCategory[],
    total: 0,
    activeCategory: null as ProjectCategory | null,
    loading: false,
  }),
  actions: {
    async fetchCategories(params?: any) {
      this.loading = true
      try {
        const res = await getProjectCategoriesApi(params)
        this.categories = res.data.data
        this.total = res.data.meta.total
      } finally {
        this.loading = false
      }
    },
    async fetchCategoryByID(id: string) {
      this.loading = true
      try {
        const res = await getProjectCategoryByIDApi(id)
        this.activeCategory = res.data.data
      } finally {
        this.loading = false
      }
    },
    async createCategory(data: any) {
      const res = await createProjectCategoryApi(data)
      return res.data.data
    },
    async updateCategory(id: string, data: any) {
      const res = await updateProjectCategoryApi(id, data)
      return res.data.data
    },
    async deleteCategory(id: string) {
      await deleteProjectCategoryApi(id)
    },
  },
})

File: backend/models/project_category.go
package models

import (
	"github.com/google/uuid"
	"time"
)

type ProjectCategory struct {
	ID        uuid.UUID `gorm:"type:uuid;primaryKey" json:"id"`
	CompanyID uuid.UUID `gorm:"type:uuid;not null;index" json:"company_id"`
	Name      string    `gorm:"type:varchar(255);not null;uniqueIndex:,composite:company_id" json:"name"`
	Code      string    `gorm:"type:varchar(50);uniqueIndex:,composite:company_id" json:"code"`
	Color     string    `gorm:"type:varchar(10)" json:"color"`
	IsActive  bool      `gorm:"not null;default:true" json:"is_active"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}
